src/rust/runtime/types/ops.rs

```diff
@@ -51,8 +47,57 @@ pub union demi_qr_value_t {
 #[repr(C)]
 pub struct demi_qresult_t {
     pub qr_opcode: demi_opcode_t,
-    pub qr_qd: c_int,
+    pub qr_qd: u32,
```

Comment:
what is the reasoning behind using u32 for queue descriptors and u64 for return values? Is it for alignment? Since the return value is a libc errno, it must also be u32.

Reply:
Here is the reasons:
- Rationale for a u32 queue descriptor: https://github.com/demikernel/demikernel/pull/427, https://github.com/demikernel/demikernel/pull/441, https://github.com/demikernel/demikernel/issues/430
- Rationale for a u64 queue token: it ensures that the structure would have the correct size and it is big enough for holding a i32. We could alternatively: (a) leave it u32, but introduce a u32 unused field for padding, which would require even more intrusive changes; or (b) leave it u32 and pack the structure, which would require massive changes to the code base.